### PR TITLE
Embeds OVAL details about test to graph

### PR DESCRIPTION
**Pull request by Honny1**
This PR embeds OVAL details about test to graph. After click on some test display OVAL details about test. Does not implement reference variables.

Fixes: #66

### CODE REVIEW

File: oval_graph/parts/footer.html [hunk 1]
```diff
@@ -1,13 +1,137 @@
...
+							break;
+						case 'exec':
+							out += "x";
+							break;
+						default:
+							// pass
+							break;
+					}
+				} else {
+					out += "-";
```
Comment (matejak) [inline oval_graph/parts/footer.html]: This is not so simple - in OVAL, we may either say that the state is `uread` on or off, which you handle well. But if we don't care, you will visualize it the same as if we want it off. I suggest using e.g. `?` to indicate the "we don't care" option.

File: oval_graph/parts/footer.html [hunk 2]
```diff
@@ -1,13 +1,137 @@
...
+			var index_special_char = str.indexOf('@');
+			var text = str.substring(0, index_special_char != -1 ? index_special_char : str.length);
+			var text_with_spaces = text.replace("_", " ");
+			return text_with_spaces.charAt(0).toUpperCase() + text_with_spaces.slice(1);
+		}
+
+		function create_table(type, table_object) {
+			switch (type) {
+				case 'file_object':
+					table_object = prepare_file_object(table_object);
```
Comment (matejak) [inline oval_graph/parts/footer.html]: Try to check out if you can detect whether a function `prepare_<type>` exists, and if yes, apply it on the table_object. That way, you will get rid of the `switch` and you make addition of "js visualization plugins" easier.
If it's too complicated, let's do it in another PR.

File: tests/test_graph.py [hunk 3]
```diff
@@ -52,78 +52,78 @@ def bad_tree():
          t
     """
     t = OvalNode(
-        1, "value", "true", False, None, None,  [
+        1, "value", "true", False, None, None, None, [
```
Comment (matejak) [inline tests/test_graph.py]: General remark: You seem to have to update tests quite a lot if you make a change in API. In cases like this, it would be worth thinking to use named arguments and arguments with default values, so you don't have to alter all those function invocations except cases when you want to pass something else than `None`.
Named arguments help against confusion what has been passed and what is supplied by default.

Comment (Honny1) [reply tests/test_graph.py]: I'll do it in another PR.

**Comment by matejak** [comment]
There is an issue with a file permission-related rule `file_permissions_library_dirs`:

- The OVAL specifies (searches for) objects that have `gwrite` and `owrite` set.
- The summary indicates that it specifies objects with all permissions unset
image

**Comment by matejak** [comment]
Great enhancement! Some issues may pop up, as introduction of OVAL objects is a huge thing, but they can be addressed in subsequent PRs.
